src/profile/v100/app_preferences/resource_preferences/resource_preferences.rs

```diff
@@ -0,0 +1,268 @@
+use crate::prelude::*;
+use core::hash::Hash;
+use std::hash::Hasher;
+
+/// User off-ledger preferences regarding resources.
+#[derive(
+    Debug,
+    Default,
+    Serialize,
+    Deserialize,
+    PartialEq,
+    Eq,
+    Clone,
+    derive_more::Display,
+    uniffi::Record,
+)]
+#[serde(rename_all = "camelCase")]
+#[display("{}", self.description())]
+pub struct ResourcePreferences {
+    /// A dictionary detailing the user preferences for fungible resources.
+    #[serde(default)]
+    pub fungible: HashMap<ResourceAddress, EntityFlags>,
```

Comment:
so it would be something like:

```rust
pub enum AssetVisibility {
    Hidden, 
    // Just an example of a potential future variant
    ConditionallyHiddenIfValueIsLessThan1000Usd
}
pub struct AssetPreference {
    address: AssetAddress, // fungible | nonFungible | poolUnit

    /// This does NOT allow for other flags, it is a mutually exclusive state.
    visibility: AssetVisibility
}

struct AssetPreference {
   assets: IndexMap<AssetAddress, AssetPreference>
}
```

Advantages:
* This retains insertion order of all preferences, "globally" (if ever interesting). 
* Is more compact
* If we want preferences for more than visibility, we add a new enum for that, and a new field "next to" `visibility`.

This is also future proof, since we can always add more enum variants to `AssetVisibility` and we can alway add new enum types and **fiels** to `AssetPreference`. So... should work?

Reply:
Agree that maybe `EntityFlags` aren't a great representation of "flags for an address". However, the reason I used them, is because if we ever extend the configuration of what is defined for entities, it probably applies for assets as well. 

Of course, this is all based on assumptions that we don't know yet how will they turn out. In your `ConditionallyHiddenIfValueIsLessThan1000Usd`, then it definitely makes sense to have the new resource configuration using a new preference definition (as the one you suggested). On my case, I thought a good example could have been "marking a resource/persona/account as favorite". In such case, a new `EntityFlag::IsFavorite` would fit for all of them.

Reply:
Shared variants is not the same as semantically the same enum :) they should be able to evolve independently of each other. Also `DeletedByUser` is the intention of account, they dont want it. Initially we gave the button the name Delete, then changed to Hide. Whereas "Hide asset" feature truely is a hide, since it is stored in the immutable Ledger - thus never can be deleted! So it should be named `WhatIsTheNameOfTheEnum::Hidden` and not `::Deleted`.

Reply:
but with that argument hidden accounts/personas should also be named `::Hidden`since they aren't deleted from Ledger